diff -Nur glib-2.54.2.orig/gio/gdbusaddress.c glib-2.54.2/gio/gdbusaddress.c
--- glib-2.54.2.orig/gio/gdbusaddress.c	2017-08-07 16:58:31.000000000 +0200
+++ glib-2.54.2/gio/gdbusaddress.c	2017-12-19 22:05:14.000000000 +0100
@@ -1435,7 +1435,6 @@
 {
   HANDLE autolaunch_mutex, init_mutex;
   char *address = NULL;
-  wchar_t gio_path[MAX_PATH+1+200];
 
   autolaunch_mutex = acquire_mutex (DBUS_AUTOLAUNCH_MUTEX);
 
@@ -1446,8 +1445,11 @@
 
   release_mutex (init_mutex);
 
+#ifdef DLL_EXPORT
+
   if (address == NULL)
     {
+      wchar_t gio_path[MAX_PATH+1+200];
       gio_path[MAX_PATH] = 0;
       if (GetModuleFileNameW (_g_io_win32_get_module (), gio_path, MAX_PATH))
 	{
@@ -1490,6 +1492,8 @@
 	}
     }
 
+#endif
+
   release_mutex (autolaunch_mutex);
 
   if (address == NULL)
