diff -bur coreclr/functions.cmake coreclr/functions.cmake
--- coreclr/functions.cmake	2017-02-09 16:48:09.009209414 +0100
+++ coreclr/functions.cmake	2017-02-09 16:52:53.303586505 +0100
@@ -117,53 +117,9 @@
   endif(MSVC)
 endfunction()
 
-function(strip_symbols targetName outputFilename)
-  if (CLR_CMAKE_PLATFORM_UNIX)
-    if (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELEASE)
-
-      # On the older version of cmake (2.8.12) used on Ubuntu 14.04 the TARGET_FILE
-      # generator expression doesn't work correctly returning the wrong path and on
-      # the newer cmake versions the LOCATION property isn't supported anymore.
-      if (CMAKE_VERSION VERSION_EQUAL 3.0 OR CMAKE_VERSION VERSION_GREATER 3.0)
-          set(strip_source_file $<TARGET_FILE:${targetName}>)
-      else()
-          get_property(strip_source_file TARGET ${targetName} PROPERTY LOCATION)
-      endif()
-
-      if (CMAKE_SYSTEM_NAME STREQUAL Darwin)
-        set(strip_destination_file ${strip_source_file}.dwarf)
-
-        add_custom_command(
-          TARGET ${targetName}
-          POST_BUILD
-          VERBATIM 
-          COMMAND ${DSYMUTIL} --flat --minimize ${strip_source_file}
-          COMMAND ${STRIP} -S ${strip_source_file}
-          COMMENT Stripping symbols from ${strip_source_file} into file ${strip_destination_file}
-        )
-      else (CMAKE_SYSTEM_NAME STREQUAL Darwin)
-        set(strip_destination_file ${strip_source_file}.dbg)
-
-        add_custom_command(
-          TARGET ${targetName}
-          POST_BUILD
-          VERBATIM 
-          COMMAND ${OBJCOPY} --only-keep-debug ${strip_source_file} ${strip_destination_file}
-          COMMAND ${OBJCOPY} --strip-debug ${strip_source_file}
-          COMMAND ${OBJCOPY} --add-gnu-debuglink=${strip_destination_file} ${strip_source_file}
-          COMMENT Stripping symbols from ${strip_source_file} into file ${strip_destination_file}
-        )
-      endif (CMAKE_SYSTEM_NAME STREQUAL Darwin)
-
-      set(${outputFilename} ${strip_destination_file} PARENT_SCOPE)
-    endif(UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELEASE)
-  endif(CLR_CMAKE_PLATFORM_UNIX)
-endfunction()
-
 function(install_clr targetName)
   list(FIND CLR_CROSS_COMPONENTS_LIST ${targetName} INDEX)  
   if (NOT DEFINED CLR_CROSS_COMPONENTS_LIST OR NOT ${INDEX} EQUAL -1)  
-    strip_symbols(${targetName} strip_destination_file) 
     # On the older version of cmake (2.8.12) used on Ubuntu 14.04 the TARGET_FILE  
     # generator expression doesn't work correctly returning the wrong path and on  
     # the newer cmake versions the LOCATION property isn't supported anymore.  
diff -bur coreclr/src/.nuget/Microsoft.NETCore.ILAsm/rhel/Microsoft.NETCore.ILAsm.pkgproj coreclr/src/.nuget/Microsoft.NETCore.ILAsm/rhel/Microsoft.NETCore.ILAsm.pkgproj
--- coreclr/src/.nuget/Microsoft.NETCore.ILAsm/rhel/Microsoft.NETCore.ILAsm.pkgproj	2017-02-09 16:43:17.978829906 +0100
+++ coreclr/src/.nuget/Microsoft.NETCore.ILAsm/rhel/Microsoft.NETCore.ILAsm.pkgproj	2017-02-09 16:53:51.676458308 +0100
@@ -15,7 +15,6 @@
     </File>
   </ItemGroup>
   <ItemGroup Condition="'$(__BuildType)' == 'Release'">
-    <ArchitectureSpecificNativeSymbol Include="@(NativeSplittableBinary -> '%(Identity).dbg')" />
     <AdditionalLibPackageExcludes Include="%2A%2A\%2A.dbg" />
     <AdditionalSymbolPackageExcludes Include="%2A%2A\%2A.so" />
     <AdditionalSymbolPackageExcludes Include="%2A%2A\%2A.dll" />
diff -bur coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/rhel/Microsoft.NETCore.ILDAsm.pkgproj coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/rhel/Microsoft.NETCore.ILDAsm.pkgproj
--- coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/rhel/Microsoft.NETCore.ILDAsm.pkgproj	2017-02-09 16:43:17.979829887 +0100
+++ coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/rhel/Microsoft.NETCore.ILDAsm.pkgproj	2017-02-09 16:54:22.136862080 +0100
@@ -15,7 +15,6 @@
     </File>
   </ItemGroup>
   <ItemGroup Condition="'$(__BuildType)' == 'Release'">
-    <ArchitectureSpecificNativeSymbol Include="@(NativeSplittableBinary -> '%(Identity).dbg')" />
     <AdditionalLibPackageExcludes Include="%2A%2A\%2A.dbg" />
     <AdditionalSymbolPackageExcludes Include="%2A%2A\%2A.so" />
     <AdditionalSymbolPackageExcludes Include="%2A%2A\%2A.dll" />
diff -bur coreclr/src/.nuget/Microsoft.NETCore.Jit/rhel/Microsoft.NETCore.Jit.pkgproj coreclr/src/.nuget/Microsoft.NETCore.Jit/rhel/Microsoft.NETCore.Jit.pkgproj
--- coreclr/src/.nuget/Microsoft.NETCore.Jit/rhel/Microsoft.NETCore.Jit.pkgproj	2017-02-09 16:43:17.979829887 +0100
+++ coreclr/src/.nuget/Microsoft.NETCore.Jit/rhel/Microsoft.NETCore.Jit.pkgproj	2017-02-09 16:54:47.837318530 +0100
@@ -15,7 +15,6 @@
     </File>
   </ItemGroup>
   <ItemGroup Condition="'$(__BuildType)' == 'Release'">
-    <ArchitectureSpecificNativeSymbol Include="@(NativeSplittableBinary -> '%(Identity).dbg')" />
     <AdditionalLibPackageExcludes Include="%2A%2A\%2A.dbg" />
     <AdditionalSymbolPackageExcludes Include="%2A%2A\%2A.so" />
     <AdditionalSymbolPackageExcludes Include="%2A%2A\%2A.dll" />
diff -bur coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/rhel/Microsoft.NETCore.Runtime.CoreCLR.pkgproj coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/rhel/Microsoft.NETCore.Runtime.CoreCLR.pkgproj
--- coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/rhel/Microsoft.NETCore.Runtime.CoreCLR.pkgproj	2017-02-09 16:43:20.140788221 +0100
+++ coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/rhel/Microsoft.NETCore.Runtime.CoreCLR.pkgproj	2017-02-09 16:55:29.787496952 +0100
@@ -40,7 +40,6 @@
     </File>
   </ItemGroup>
   <ItemGroup Condition="'$(__BuildType)' == 'Release'">
-    <ArchitectureSpecificNativeSymbol Include="@(NativeSplittableBinary -> '%(Identity).dbg')" />
     <AdditionalLibPackageExcludes Include="%2A%2A\%2A.dbg" />
     <AdditionalSymbolPackageExcludes Include="%2A%2A\%2A.so" />
     <AdditionalSymbolPackageExcludes Include="%2A%2A\%2A.dll" />
