--- gdk-pixbuf/Makefile.in.orig	2010-10-07 21:46:29.959447520 +0200
+++ gdk-pixbuf/Makefile.in	2010-10-07 21:48:49.112895455 +0200
@@ -1344,7 +1344,7 @@
 	$(gdk_pixbuf_symbols)			\
 	$(gdiplus_ldflag)
 
-libgdk_pixbuf_2_0_la_LIBADD = pixops/libpixops.la $(builtin_objs) $(GDK_PIXBUF_DEP_LIBS) $(libole32)
+libgdk_pixbuf_2_0_la_LIBADD = pixops/libpixops.la $(builtin_objs) $(GDK_PIXBUF_DEP_LIBS) $(libole32) -lintl
 libgdk_pixbuf_2_0_la_DEPENDENCIES = pixops/libpixops.la $(builtin_objs) $(gdk_pixbuf_def) $(gdk_pixbuf_win32_res) $(libgdiplus)
 gdk_pixbuf_headers = \
 	gdk-pixbuf.h			\
