diff -Nur glib-2.50.3.orig/gio/Makefile.am glib-2.50.3/gio/Makefile.am
--- glib-2.50.3.orig/gio/Makefile.am	2017-02-13 16:15:29.000000000 +0100
+++ glib-2.50.3/gio/Makefile.am	2017-03-04 20:03:04.000000000 +0100
@@ -775,6 +775,10 @@
 	$(top_builddir)/glib/libglib-2.0.la 		\
 	$(NULL)
 
+if OS_COCOA
+glib_compile_resources_LDFLAGS = -Wl,-framework,Foundation -Wl,-framework,AppKit
+endif
+
 glib_compile_resources_SOURCES = \
 	gvdb/gvdb-format.h		\
 	gvdb/gvdb-builder.h		\
@@ -788,6 +792,10 @@
 	$(top_builddir)/glib/libglib-2.0.la		\
 	$(NULL)
 
+if OS_COCOA
+glib_querymodules_LDFLAGS = -Wl,-framework,Foundation -Wl,-framework,AppKit
+endif
+
 gconstructor_as_data.h: $(top_srcdir)/glib/gconstructor.h data-to-c.pl
 	$(AM_V_GEN) $(srcdir)/data-to-c.pl $(top_srcdir)/glib/gconstructor.h gconstructor_code > $@.tmp && mv $@.tmp $@
 
@@ -799,6 +807,10 @@
 	gvdb/gvdb-builder.c		\
 	glib-compile-schemas.c
 
+if OS_COCOA
+glib_compile_schemas_LDFLAGS = -Wl,-framework,Foundation -Wl,-framework,AppKit
+endif
+
 gsettings_LDADD = libgio-2.0.la 			\
 	$(top_builddir)/gobject/libgobject-2.0.la	\
 	$(top_builddir)/gmodule/libgmodule-2.0.la 	\
@@ -806,6 +818,10 @@
 	$(NULL)
 gsettings_SOURCES = gsettings-tool.c
 
+if OS_COCOA
+gsettings_LDFLAGS = -Wl,-framework,Foundation -Wl,-framework,AppKit
+endif
+
 schemadir = $(datadir)/glib-2.0/schemas
 dist_schema_DATA = gschema.dtd
 
@@ -857,6 +873,10 @@
 	$(top_builddir)/glib/libglib-2.0.la		\
 	$(NULL)
 
+if OS_COCOA
+gdbus_LDFLAGS = -Wl,-framework,Foundation -Wl,-framework,AppKit
+endif
+
 if OS_UNIX
 # ------------------------------------------------------------------------
 # gapplication(1) tool
@@ -867,6 +887,11 @@
 	$(top_builddir)/gmodule/libgmodule-2.0.la 	\
 	$(top_builddir)/glib/libglib-2.0.la		\
 	$(NULL)
+
+if OS_COCOA
+gapplication_LDFLAGS = -Wl,-framework,Foundation -Wl,-framework,AppKit
+endif
+
 endif
 
 completiondir = $(datadir)/bash-completion/completions
@@ -889,6 +914,10 @@
 	$(top_builddir)/glib/libglib-2.0.la		\
 	$(LIBELF_LIBS)
 
+if OS_COCOA
+gresource_LDFLAGS = -Wl,-framework,Foundation -Wl,-framework,AppKit
+endif
+
 # ------------------------------------------------------------------------
 # gio tool
 
