diff -rupN --no-dereference gst-plugins-base-1.26.3/gst-libs/gst/audio/gstaudioutilsprivate.c gst-plugins-base-1.26.3-new/gst-libs/gst/audio/gstaudioutilsprivate.c
--- gst-plugins-base-1.26.3/gst-libs/gst/audio/gstaudioutilsprivate.c	2025-06-26 22:25:24.000000000 +0200
+++ gst-plugins-base-1.26.3-new/gst-libs/gst/audio/gstaudioutilsprivate.c	2025-06-29 15:42:20.836905166 +0200
@@ -301,9 +301,9 @@ __gst_audio_set_thread_priority (gpointe
 
 #ifdef G_OS_WIN32
   /* This is only used from ringbuffer thread functions */
-  *handle = (gpointer)
-      _gst_audio_avrt_tbl.AvSetMmThreadCharacteristics (TEXT ("Pro Audio"),
-      &taskIndex);
+  typedef HANDLE (WINAPI *AvSetMmThreadCharacteristics_t)(LPCSTR, LPDWORD);
+  AvSetMmThreadCharacteristics_t AvSetMmThreadCharacteristicsFn = (AvSetMmThreadCharacteristics_t)_gst_audio_avrt_tbl.AvSetMmThreadCharacteristics;
+  *handle = (gpointer) AvSetMmThreadCharacteristicsFn (TEXT ("Pro Audio"), &taskIndex);
   if (*handle == 0) {
     gchar *errorMsg = g_win32_error_message (GetLastError ());
 
@@ -333,7 +333,9 @@ __gst_audio_restore_thread_priority (gpo
   if (!handle)
     return FALSE;
 
-  return _gst_audio_avrt_tbl.AvRevertMmThreadCharacteristics ((HANDLE) handle);
+  typedef BOOL (WINAPI *AvRevertMmThreadCharacteristics_t)(HANDLE);
+  AvRevertMmThreadCharacteristics_t AvRevertMmThreadCharacteristicsFn = (AvRevertMmThreadCharacteristics_t)_gst_audio_avrt_tbl.AvRevertMmThreadCharacteristics;
+  return AvRevertMmThreadCharacteristicsFn ((HANDLE) handle);
 #else
   return TRUE;
 #endif
