diff -rupN icu/source/configure icu-new/source/configure
--- icu/source/configure	2023-12-11 23:27:41.000000000 +0100
+++ icu-new/source/configure	2024-07-26 16:00:43.039446191 +0200
@@ -6134,7 +6134,7 @@ if [ "$GXX" = yes ]; then
     # and check that the compiler still works.
     if ! echo "$CXXFLAGS" | grep '\-std=' >/dev/null 2>&1; then
         OLD_CXXFLAGS="${CXXFLAGS}"
-        CXXFLAGS="$CXXFLAGS -std=c++11"
+        CXXFLAGS="$CXXFLAGS -std=gnu++11"
         { $as_echo "$as_me:${as_lineno-$LINENO}: checking if we have a C++11 compiler" >&5
 $as_echo_n "checking if we have a C++11 compiler... " >&6; }
         cat confdefs.h - <<_ACEOF >conftest.$ac_ext
diff -rupN icu/source/configure.ac icu-new/source/configure.ac
--- icu/source/configure.ac	2023-12-11 23:27:41.000000000 +0100
+++ icu-new/source/configure.ac	2024-07-26 15:59:54.454490711 +0200
@@ -519,7 +519,7 @@ if [[ "$GXX" = yes ]]; then
     # and check that the compiler still works.
     if ! echo "$CXXFLAGS" | grep '\-std=' >/dev/null 2>&1; then
         OLD_CXXFLAGS="${CXXFLAGS}"
-        CXXFLAGS="$CXXFLAGS -std=c++11"
+        CXXFLAGS="$CXXFLAGS -std=gnu++11"
         AC_MSG_CHECKING([[if we have a C++11 compiler]])
         AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]],[[]])],[cxx11_okay=yes],[cxx11_okay=no])
         AC_MSG_RESULT($cxx11_okay)
