diff -Nur ige-mac-bundler-0.5.orig/bundler/launcher.sh ige-mac-bundler-0.5/bundler/launcher.sh
--- ige-mac-bundler-0.5.orig/bundler/launcher.sh	2008-05-10 19:52:31.000000000 +0200
+++ ige-mac-bundler-0.5/bundler/launcher.sh	2025-06-03 14:58:03.353798585 +0200
@@ -29,11 +29,6 @@
 export GTK_EXE_PREFIX="$bundle_res"
 export GTK_PATH="$bundle_res"
 
-export GTK2_RC_FILES="$bundle_etc/gtk-2.0/gtkrc"
-export GTK_IM_MODULE_FILE="$bundle_etc/gtk-2.0/gtk.immodules"
-export GDK_PIXBUF_MODULE_FILE="$bundle_etc/gtk-2.0/gdk-pixbuf.loaders"
-export PANGO_RC_FILE="$bundle_etc/pango/pangorc"
-
 # We need a UTF-8 locale.
 lang=`defaults read .GlobalPreferences AppleLocale 2>/dev/null`
 if test "$?" != "0"; then
