diff -rupN --no-dereference jasper-version-4.2.4/src/libjasper/CMakeLists.txt jasper-version-4.2.4-new/src/libjasper/CMakeLists.txt
--- jasper-version-4.2.4/src/libjasper/CMakeLists.txt	2024-04-27 19:53:19.000000000 +0200
+++ jasper-version-4.2.4-new/src/libjasper/CMakeLists.txt	2024-07-07 16:15:08.485210264 +0200
@@ -188,9 +188,12 @@ if(MSVC)
 	target_compile_definitions(libjasper PRIVATE _CRT_NONSTDC_NO_DEPRECATE)
 endif()
 
-if(UNIX)
+if(UNIX OR MINGW)
 	set_target_properties(libjasper PROPERTIES
 	  VERSION ${JAS_SO_NAME} SOVERSION ${JAS_SO_VERSION})
+	if(MINGW AND JAS_ENABLE_SHARED)
+		set_target_properties(libjasper PROPERTIES SUFFIX "-${JAS_SO_VERSION}${CMAKE_SHARED_LIBRARY_SUFFIX}")
+	endif(MINGW AND JAS_ENABLE_SHARED)
 endif()
 
 if(JAS_ENABLE_SHARED)
