diff -rupN --no-dereference jasper-version-4.2.4/src/libjasper/include/jasper/jas_image.h jasper-version-4.2.4-new/src/libjasper/include/jasper/jas_image.h
--- jasper-version-4.2.4/src/libjasper/include/jasper/jas_image.h	2024-04-27 19:53:19.000000000 +0200
+++ jasper-version-4.2.4-new/src/libjasper/include/jasper/jas_image.h	2024-07-07 16:15:08.524209309 +0200
@@ -861,9 +861,9 @@ int jp2_validate(jas_stream_t *in);
 
 #if defined(JAS_INCLUDE_JPC_CODEC)
 /* Format-dependent operations for JPEG-2000 code stream support. */
-//JAS_EXPORT
+JAS_EXPORT
 jas_image_t *jpc_decode(jas_stream_t *in, const char *optstr);
-//JAS_EXPORT
+JAS_EXPORT
 int jpc_encode(jas_image_t *image, jas_stream_t *out, const char *optstr);
 //JAS_EXPORT
 int jpc_validate(jas_stream_t *in);
diff -rupN --no-dereference jasper-version-4.2.4/src/libjasper/jp2/jp2_cod.h jasper-version-4.2.4-new/src/libjasper/jp2/jp2_cod.h
--- jasper-version-4.2.4/src/libjasper/jp2/jp2_cod.h	2024-04-27 19:53:19.000000000 +0200
+++ jasper-version-4.2.4-new/src/libjasper/jp2/jp2_cod.h	2024-07-07 16:15:08.524209309 +0200
@@ -288,10 +288,10 @@ typedef struct jp2_boxinfo_s {
 * Box class.
 \******************************************************************************/
 
-jp2_box_t *jp2_box_create(int type);
-void jp2_box_destroy(jp2_box_t *box);
-jp2_box_t *jp2_box_get(jas_stream_t *in);
-int jp2_box_put(jp2_box_t *box, jas_stream_t *out);
+JAS_DLLEXPORT jp2_box_t *jp2_box_create(int type);
+JAS_DLLEXPORT void jp2_box_destroy(jp2_box_t *box);
+JAS_DLLEXPORT jp2_box_t *jp2_box_get(jas_stream_t *in);
+JAS_DLLEXPORT int jp2_box_put(jp2_box_t *box, jas_stream_t *out);
 
 JAS_ATTRIBUTE_CONST
 static inline uint_least8_t JP2_DTYPETOBPC(uint_least8_t dtype)
